sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -105,6 +105,8 @@ case class DataSource(
       case _ => cls
     }
   }
+  private def providingInstance = providingClass.getConstructor().newInstance()
```

Comment:
This should be declared like a method for clarity, with a return type. Also add a new line above. I'd use `()` in the declaration and invocations for clarity too.

Reply:
If we add a return type, only Any could use here. This method is modified by private, so whether the return type can be omitted?

Reply:
The type would be something like `_ <: DataSource`, not `Any`, but, OK.

Reply:
I used `Any` temporarily. Because the type of `providingClass` is `Class[_]`. If we really want use `_ <: DataSourceRegister`, we have to adust the implement of `DataSource.lookupDataSource`.